BLSQ/iaso
# IA-1511 Add Daily frequency on Forms

Allow daily period on Form.
It was mainly so it could be listed on the mobile application. But we had to adapt on the frontend too so it could be displayed and used properly and avoid crashes (see changes).
So it's quite a big PR

## Self proof reading checklist

- [] Are there enough tests

## Changes

- Add option in the Form settings
- Adapt the Create or Reassign submission dialog to accept day. Rewrote most of it to use PeriodPicker like we do for the rest and not load before opened using Son method
- Adapt the period filter on Instance (front and back)
- Clean some code, add plenty of Typescript

## How to test
Create form with daily period
Add a form version, fill a form.
Change it's period in the instance detail
Change the period type on the Form see if everything still works
Create a few form, try filtering.


## Notes
- When we search the year period, we usually include the result for Semester, Month, etc.. as a compromise for now to limit the performance impact we don't do that for day period at the moment. I fell this is reasonable for now as the PR is already quite long but we should improve it in the future.
- Completeness (1) table has not been adapted as discussed with the geo team.  Didn't check if completness 2 table is working.
- I didn't test it if it worked on mobile. Benjamin said it should but it would be better to validate once it is in staging.
